pkg/querier/querier.go

```diff
@@ -27,6 +27,9 @@ type Config struct {
 	MaxSamples               int
 	IngesterMaxQueryLookback time.Duration
 
+	// IngesterMetricMinAge minimum time between a metric being ingested and saved to the store.
+	IngesterMetricMinAge time.Duration `yaml:"ingester_metric_min_age"`
```

Comment:
How does this option relates to ingester's `MaxChunkIdle` and `MaxStaleChunkIdle`? The problem I see is that if this option is > than `min(MaxChunkIdle, MaxStaleChunkIdle)` then you may end up in weird bugs hiding series which have been flushed to the storage.

If I'm not missing anything, we shouldn't mine Cortex correctness in any way and we should protect from it.

Reply:
I don't think it changes that relation at all right? If a user had set `MinChunkAge` to be `1h` but set `MaxChunkIdle` to `1s` it could leave a gap during query time. I'm not sure it's a bug but a known configuration pattern.

Reply:
It looks a bug to me (given it's a quite tricky thing even to correctly configure), but I would be happy to be contradicted. @gouthamve @bboreham any thought?

Reply:
I think it's a combination of `min(MaxChunkIdle, MaxStaleChunkIdle)` + `-ingester.retain-period`. As long as we have the retain period, we don't need to hit the store. Though I agree its hard to configure and get right.